src/zenml/utils/exception_utils.py

```diff
@@ -91,3 +93,41 @@ def collect_exception_information(
         traceback=tb_bytes.decode(errors="ignore"),
         step_code_line=line_number,
     )
+
+
+class ContextReraise(ContextDecorator):
+    def __init__(
+        self,
+        source_exceptions: list[Type[BaseException]],
+        target_exception: Type[BaseException],
+        message: str,
+        propagate_traceback: bool = True,
+    ) -> None:
+        self._source_exceptions = source_exceptions
+        self._target_exception = target_exception
+        self._message = message
```

Comment:
I guess to simplify this, we could just pass an instance of the exception here instead of the class and message? That would additionally also allow some exceptions which can/need to be instantiated with multiple arguments.